signac/contrib/project.py

```diff
@@ -1544,20 +1539,32 @@ def init_project(cls, *args, root=None, workspace=None, make_dir=True, **kwargs)
                 raise TypeError(
                     f"init_project() got an unexpected keyword argument '{next(iter(kwargs))}'"
                 )
-            if name is not None:
-                assert int(version.parse(__version__).public.split(".")[0]) < 3
-                warnings.warn(
-                    "Project names were removed in signac 2.0. If your project name contains "
-                    "important information, consider storing it in the project document instead.",
-                    FutureWarning,
-                )
+
+        if name is not None:
+            assert version.parse(__version__) < version.parse("3.0.0")
+            warnings.warn(
+                "Project names were removed in signac 2.0. If you intended to call "
+                "`init_project` with a root directory as the sole positional argument, please "
+                f"provide it as a keyword argument: `init_project(root={name})`. If your "
+                "project name contains important information, consider storing it in the "
+                "project document instead. The name provided will be stored in the project "
+                f"document with the key `{name_key}`. Calling `init_project` with a name will "
+                "become an error in signac 3.0.",
+                FutureWarning,
+            )
 
         if root is None:
             root = os.getcwd()
         try:
             project = cls.get_project(root=root, search=False)
+            existing_name = project.doc.get(name_key)
+            if name is not None and name != existing_name:
+                raise ValueError(
+                    "The name provided to `init_project` does not match the existing "
+                    f"project document in which {name_key}={existing_name}."
+                )
         except LookupError:
-            name = "project"
+            name = name or "project"
```

Comment:
Why don't we use `None` as the default/fallback instead of the literal `"project"`?

Reply:
Good point. No reason. Changed it.